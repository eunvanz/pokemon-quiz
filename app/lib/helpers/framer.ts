import { Variants } from 'framer-motion'

export const motionVariants: Variants = {
  vibe: {
    translateX: [0, -4, 0, 4, 0],
    transition: {
      repeat: 2,
      duration: 0.1,
    },
  },
}
