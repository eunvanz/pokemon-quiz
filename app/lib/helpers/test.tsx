import CommonProvider from '@/app/components/common-provider'
import { PropsWithChildren } from 'react'
import api from '../api'

export interface TestProviderProps {
  api?: Partial<typeof api>
}

export const TestProvider: React.FC<PropsWithChildren<TestProviderProps>> = ({
  children,
  api = {},
}) => <CommonProvider api={api}>{children}</CommonProvider>
