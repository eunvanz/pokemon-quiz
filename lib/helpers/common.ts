export const checkIsSSR = () => {
  return typeof window === 'undefined'
}
