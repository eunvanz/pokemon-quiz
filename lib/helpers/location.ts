import { countries, zones } from 'moment-timezone/data/meta/latest.json'

export const getTimeZoneToCountryMap = () => {
  const timeZoneToCountryMap: Record<string, { name: string; abbr: string }> =
    {}
  Object.keys(zones).forEach((zone) => {
    const cityArr = zone.split('/')
    const city = cityArr[cityArr.length - 1]
    // @ts-ignore
    timeZoneToCountryMap[city] = countries[zones[zone].countries[0]]
  })
  return timeZoneToCountryMap
}

export const getUserLocationFromTimeZone = () => {
  const timeZone = Intl?.DateTimeFormat().resolvedOptions().timeZone
  if (timeZone) {
    const tzArr = timeZone.split('/')
    const region = tzArr[0]
    const city = tzArr[tzArr.length - 1]
    const { name, abbr } = getTimeZoneToCountryMap()[city]
    return {
      city,
      country: name,
      countryCode: abbr,
      region,
    }
  } else {
    return undefined
  }
}

export const getCountryFlagImgUrl = (countryCode: string) => {
  return `http://purecatamphetamine.github.io/country-flag-icons/3x2/${countryCode.toUpperCase()}.svg`
}
