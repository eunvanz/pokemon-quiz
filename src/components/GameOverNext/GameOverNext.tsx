import tw from "twin.macro";
import Button from "../Button";

export interface GameOverNextProps {
  onNext: VoidFunction;
}

const GameOverNext: React.FC<GameOverNextProps> = ({ onNext }) => {
  return (
    <div css={tw`flex flex-col items-center`}>
      <h1>Click a Pok√©mon to check the answer</h1>
      <h3>or</h3>
      <Button css={tw`w-full`} onClick={onNext}>
        Check your rank
      </Button>
    </div>
  );
};

export default GameOverNext;
