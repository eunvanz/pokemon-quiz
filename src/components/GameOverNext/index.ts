export { default } from "./GameOverNext";

export type { GameOverNextProps } from "./GameOverNext";
