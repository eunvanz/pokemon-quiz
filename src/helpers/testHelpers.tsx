import { Story } from "@storybook/react";
import { render, RenderOptions } from "@testing-library/react";
import { QueryClientProvider } from "react-query";
import api from "~/api";
import { ApiProvider } from "~/api/apiContext";
import { queryClient } from "./reactQuery";

/**
 * render storybook component in test code
 * @param StoryComponent
 * @param props
 * @returns
 */
export const renderStory = <T,>(
  StoryComponent: Story<Partial<T>>,
  props?: Partial<T>,
  renderOptions?: RenderOptions,
) => {
  return render(<StoryComponent {...props} />, renderOptions);
};

export interface TestProviderProps {
  api: Partial<typeof api>;
}

export const TestProvider: React.FC<TestProviderProps> = ({ children, api }) => (
  <ApiProvider api={api}>
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  </ApiProvider>
);
