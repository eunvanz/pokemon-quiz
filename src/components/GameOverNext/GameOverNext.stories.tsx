import type { ComponentStory, ComponentMeta } from "@storybook/react";
import { createStoryComponent } from "~/helpers/storybookHelpers";
import GameOverNext from "./GameOverNext";

export default {
  title: "components/GameOverNext",
  component: GameOverNext,
  args: {},
} as ComponentMeta<typeof GameOverNext>;

const Template: ComponentStory<typeof GameOverNext> = (args) => (
  <GameOverNext {...args} />
);

export const Default = createStoryComponent(Template);
