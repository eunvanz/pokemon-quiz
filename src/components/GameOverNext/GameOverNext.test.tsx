import { ComponentProps } from "react";
import { composeStories } from "@storybook/testing-react";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderStory } from "~/helpers/testHelpers";
import GameOverNext from "./GameOverNext";
import * as stories from "./GameOverNext.stories";

const { Default } = composeStories(stories);

describe("GameOverNext", () => {
  const setup = (props?: ComponentProps<typeof GameOverNext>) =>
    renderStory(Default, props);

  it("matches with previous snapshot.", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it("calls [onNext] callback correctly", () => {
    const onNext = jest.fn();
    setup({ onNext });

    const nextBtn = screen.getByRole("button", { name: /check your rank/i });

    userEvent.click(nextBtn);

    expect(onNext).toBeCalledTimes(1);
  });
});
