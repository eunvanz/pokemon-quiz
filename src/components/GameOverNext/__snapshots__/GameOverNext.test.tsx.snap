// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOverNext matches with previous snapshot. 1`] = `
<div>
  <div
    class="css-1q8p9k4-GameOverNext"
  >
    <h1>
      Click a Pok√©mon to check the answer
    </h1>
    <h3>
      or
    </h3>
    <button
      class="css-uxyg09-Button-GameOverNext"
    >
      Check your rank
    </button>
  </div>
</div>
`;
